packages/experience-builder-sdk/src/hooks/useFetchById.ts

```diff
@@ -3,25 +3,23 @@ import type { ContentfulClientApi } from 'contentful';
 import type { ExternalSDKMode } from '@contentful/experience-builder-core/types';
 import { useFetchByBase } from './useFetchByBase';
 import { fetchById } from '@contentful/experience-builder-core';
+import { useDetectEditorMode } from './useDetectEditorMode';
 
 export type UseFetchByIdArgs = {
-  mode: ExternalSDKMode;
+  /** @deprecated mode no longer used */
+  mode?: ExternalSDKMode;
   client: ContentfulClientApi<undefined>;
   id: string;
   experienceTypeId: string;
   localeCode: string;
 };
 
-export const useFetchById = ({
-  id,
-  localeCode,
-  client,
-  experienceTypeId,
-  mode,
-}: UseFetchByIdArgs) => {
+export const useFetchById = ({ id, localeCode, client, experienceTypeId }: UseFetchByIdArgs) => {
+  const isEditorMode = useDetectEditorMode();
```

Comment:
A hook doesn't persist it's internal state across different usages, meaning that the hook used here and the hook used in `ExperienceRoot` will both start with fresh initial state and write up message listeners.

This potentially could lead to an out-of-sync situation when let's say this hook will receive value true and `ExperienceRoot` will receive value false, because the message sent from the web app would be received by one of them before the other one registers an event listener.

Perhaps using zustand for it would be a more secure way

The same applies to the other places which use this hook. RequestEditorMode event is sent once and with this approach we should have a guarantee that all hooks will receive the following handshake and have their state in sync

Reply:
The hook itself has a bit of global state (thats the purpose of the __EB__ var on global) to deal with this issue. I didn't want to introduce a state lib that could potentially collide with userland code for one variable.